### PR TITLE
Fix search for build items

### PR DESCRIPTION
**Pull request by yrodiere**
Try searching for `NativeImageBuildItem`. Before, it didn't match https://quarkus.io/guides/all-builditems . Now it does.

Best reviewed commit by commit, to understand why each change was necessary.

I ended up removing the field(s) specific to config properties, because they caused problems with relevance and highlighting; see last commit.
It also turns out that removing that field and raising the max_gram for all autocomplete fields, plus the other changes, actually makes indexes *smaller* (832.2MB => 568.9MB). Probably because that removed field was very heavy? I don't know, but that's nice.

### CODE REVIEW

File: src/main/java/io/quarkus/search/app/hibernate/InputProviderHtmlBodyTextBridge.java [hunk 1]
```diff
@@ -21,13 +21,13 @@ public String toIndexedValue(InputProvider provider, ValueBridgeToIndexedValueCo
...
-                if (guideColumn != null) {
-                    content = guideColumn;
+                Element toc = content.selectFirst(".tocwrapper");
+                if (toc != null) {
+                    // If the guide contains a TOC, there's not much point in indexing it.
+                    // We index its sibling, the actual content.
+                    content = toc.siblingElements().stream().filter(e -> e != toc).findFirst()
+                            // If we don't find it, we fall back to indexing everything, including the toc.
+                            .orElse(content);
                 }
```
Comment (marko-bekhta) [inline src/main/java/io/quarkus/search/app/hibernate/InputProviderHtmlBodyTextBridge.java:31]:
```suggestion
                if (toc != null) {
                    // If the guide contains a TOC, there's not much point in indexing it.
                    // Just remove it from the parsed DOM
                    toc.remove();
                }
```
how about that ? 😃

File: src/main/java/io/quarkus/search/app/hibernate/InputProviderHtmlBodyTextBridge.java [hunk 2]
```diff
@@ -21,13 +21,13 @@ public String toIndexedValue(InputProvider provider, ValueBridgeToIndexedValueCo
...
             Element content = body.selectFirst(".guide");
             if (content != null) {
-                // We may be looking at a guide with/without a TOC;
-                // if it is one with the TOC, there's no uch point in indexing the TOC itself, hence we don't include it:
-
-                // column (grid items) are not present in a guide like all-config
-                Element guideColumn = content.selectFirst(".grid__item");
-                if (guideColumn != null) {
-                    content = guideColumn;
+                Element toc = content.selectFirst(".tocwrapper");
```
Comment (marko-bekhta) [inline src/main/java/io/quarkus/search/app/hibernate/InputProviderHtmlBodyTextBridge.java:24]: hmm I do not see this class in the docs... is it something new to be deployed ? for the current quarkus.io I can only find `.toc` class

**Review by marko-bekhta — COMMENTED** [review]
nice!  😃

File: src/main/java/io/quarkus/search/app/hibernate/InputProviderHtmlBodyTextBridge.java (hunk 1 above)
Comment (yrodiere) [reply src/main/java/io/quarkus/search/app/hibernate/InputProviderHtmlBodyTextBridge.java]: Makes sense!

File: src/main/java/io/quarkus/search/app/hibernate/InputProviderHtmlBodyTextBridge.java (hunk 2 above)
Comment (yrodiere) [reply src/main/java/io/quarkus/search/app/hibernate/InputProviderHtmlBodyTextBridge.java:24]: I see it... ?

image

File: src/main/java/io/quarkus/search/app/hibernate/InputProviderHtmlBodyTextBridge.java (hunk 1 above)
Comment (yrodiere) [reply src/main/java/io/quarkus/search/app/hibernate/InputProviderHtmlBodyTextBridge.java]: Seems we need to remove `.relations` too, I'll do that

File: src/main/java/io/quarkus/search/app/hibernate/InputProviderHtmlBodyTextBridge.java (hunk 2 above)
Comment (yrodiere) [reply src/main/java/io/quarkus/search/app/hibernate/InputProviderHtmlBodyTextBridge.java:24]: Anyway, with your new approach I can just remove both `.tocwrapper` and `.toc`. That should be more future-proof.

Comment (marko-bekhta) [reply src/main/java/io/quarkus/search/app/hibernate/InputProviderHtmlBodyTextBridge.java]: hmm, ok 😃 I see it now, too. Not sure what page I was looking at or maybe I should put some 👓  🤷🏻‍♂️

**Pull request merged**